### PR TITLE
Automatic deployment in npm

### LINKED ISSUE #36: Automatic deployment in npm
## I'm submitting a...

- [x] Continuous integration
- [ ] Regression (a behavior that used to work and stopped working in a new release)
- [ ] Bug report  
- [ ] Feature request
- [ ] Documentation issue or request
- [ ] Support request => Please do not submit support request here, instead see https://github.com/glpi-project/javascript-library-glpi/blob/develop/CONTRIBUTING.md#questions-or-doubts

## Expected behavior

to create a script that simplifies the library's deployment in npm, for both the beta and final versions

### PR DESCRIPTION
**Pull request by Gianfranco97**
## Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: [Conventional Commit](http://conventionalcommits.org/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Close: #36

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by Gianfranco97** [comment]
documentation about beta versions in npm 
- https://docs.npmjs.com/misc/semver#prerelease-identifiers

**Review by ajsb85 — APPROVED** [review]

**Comment by Gianfranco97** [comment]
mode used to version the beta versions:
- https://docs.npmjs.com/getting-started/using-tags

File: package.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "javascript-library-glpi",
+  "name": "@teclib/glpi-api-client",
```
Comment (ajsb85) [inline package.json:2]: Nice

**Comment by Gianfranco97** [comment]
captura de pantalla 2018-01-30 a la s 11 55 30 a m

This little friend is new, what does he want to tell me?

File: package.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
...
   "version": "0.1.0",
   "description": "GLPI API Client Library for javascript",
   "main": "lib/GlpiRestClient.js",
```
Comment (ajsb85) [inline package.json]: Could yo use another generic file name?
We will use REST and XMLRPC in the future.

**Review by ajsb85 — COMMENTED** [review]
OK

File: package.json (hunk 2 above)
Comment (Gianfranco97) [reply package.json]: Any suggestions?

**Comment by ajsb85** [comment]
The bot needs commits like this one:

image

Reference:
https://github.com/probot/dco#how-it-works

Bypass this PR but try to do it for the next commits 👍

File: package.json (hunk 2 above)
Comment (ajsb85) [reply package.json]: `GlpiApiClient.js`

**Comment by Gianfranco97** [comment]
Mmmm ... '-s' instead of '-S'

I understand

**Comment by Gianfranco97** [comment]
@ajsb85  but this information would not overload the coverage?

File: ci/scripts/build_develop.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+yarn release -t ''
+GIT_TAG=$(jq -r ".version" package.json)
+git reset HEAD^1
+
+yarn build 
+
+if [[ "$GIT_COMMIT_DESC" != *"ci(build): update library to version "* ]]
+then
+    git add lib/
+    git commit -m "ci(build): update library to version ${GIT_TAG}"
```
Comment (ajsb85) [inline ci/scripts/build_develop.sh]: `build(npm)`

File: ci/scripts/build_master.sh [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+git checkout "$CIRCLE_BRANCH"
+yarn release -t ''
+GIT_TAG=$(jq -r ".version" package.json)
+
+yarn build 
+
+if [[ "$GIT_COMMIT_DESC" != *"ci(build): update library to version "* ]]
```
Comment (ajsb85) [inline ci/scripts/build_master.sh]: build(npm)

**Review by ajsb85 — CHANGES_REQUESTED** [review]

**Comment by Gianfranco97** [comment]
@ajsb85 updated

**Review by ajsb85 — APPROVED** [review]

**Pull request merged**